packages/webapp/pages/register.tsx

```diff
@@ -39,6 +40,16 @@ export default function Register(): ReactElement {
     });
   }, []);
 
+  useEffect(() => {
+    if (!loadedUserFromCache || loadingUser || user) {
+      return;
+    }
+
+    if (!user) {
+      router?.replace((router.query.redirect_uri as string) || '/');
```

Comment:
Won't this cause a infinite loop though when it redirects people with this vague error state back to the register?